install/migration/migration.py

```diff
@@ -54,8 +54,11 @@ def bind(ldap_uri, base_dn, username, password):
         logger.error('migration unable to get base dn')
         raise IOError(errno.EIO, 'Cannot get Base DN')
     bind_dn = DN(('uid', username), ('cn', 'users'), ('cn', 'accounts'), base_dn)
+    # ldap_uri should be ldapi:// in all common cases. Enforce start_tls just
+    # in case it's a plain LDAP connection.
+    start_tls = ldap_uri.startswith('ldap://')
     try:
-        conn = ipaldap.LDAPClient(ldap_uri)
+        conn = ipaldap.LDAPClient(ldap_uri, start_tls=start_tls)
```

Comment:
ldap_uri in this case comes from api.env.ldap_uri which is going to be an ldapi uri. start_tls will be True. I don't think this will cause problems but I haven't tested it, have you?

Reply:
Are you also seeing an older version of the patch here? The new code is ``start_tls = ldap_uri.startswith('ldap://')``, so only ``start_tls=True`` for ``ldap`` URI but not for ``ldapi`` and ``ldaps``.

Reply:
You're right, my mistake.